src/middleware/next.rs

```diff
@@ -4,15 +4,18 @@ use super::ArcMiddleware;
 use crate::{BoxFuture, Request, Response, Result};
 
 pub struct Next<State = ()> {
-    stack: VecDeque<ArcMiddleware<State>>,
+    #[allow(clippy::box_collection)]
+    stack: Box<VecDeque<ArcMiddleware<State>>>,
```

Comment:
While boxing a collection can be redundant, I suspect that reducing the size of `Next` from `32` to `8` will result in better overall performance. My reasoning is that it's more important for `Next` to efficiently move between middleware rather than attempting to reduce the indirection introduced by dereferencing the pointer in `Next::call`.